lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -92,6 +93,22 @@ class ReplicationConfiguration {
     }
 
     /**
+     * Generate a 48-character alphanumeric, unique ID for a rule
+     * @return {string} - The unique rule ID
+     */
+    _createUniqueID() {
+        // If no ID is given in the replication configuration, we create it.
+        const id = Buffer.from(UUID.v4()).toString('base64');
+        // Each ID in a list of rules must be unique. If the generated ID
+        // already exists, call recursively until a unique ID is generated.
+        if (this._configIDs.includes(id)) {
+            return this._createUniqueID();
```

Comment:
A recursion should have an exit condition. There is no way that `uuid` will generate non-unique id on retry but for safety I would say limit it to 5/10 tries. If it fails to generate a unique id, we will take the non-unique id as technically this id doesn't serve any functional purpose for us, yet.